caikit/core/task.py

```diff
@@ -169,9 +169,17 @@ def validate_run_signature(
         ).items():
             signature_type = signature.parameters[parameter_name]
             if parameter_type != signature_type:
-                if typing.get_origin(
-                    signature_type
-                ) == typing.Union and parameter_type in typing.get_args(signature_type):
+                if typing.get_origin(signature_type) == typing.Union and (
+                    # Either our parameter type is not a union & is part of the union signature
+                    parameter_type in typing.get_args(signature_type)
+                    # Or our parameter type is a union that's a subset of the union signature
+                    or (
+                        typing.get_origin(parameter_type) == typing.Union
```

Comment:
Do we need to handle `Annotated` here at all? I guess we probably need to unwrap that before this loop? Probably not part of this PR though.

Reply:
NOTE for future: The `validate_run_signature` is only called at decoration time, so while this additional logic may _slightly_ slow down `import` time, it will have no impact at runtime.

Reply:
Good question - probably, although currently we don't us annotated explicitly in our task definitions and module definitions, right? When are we expected to have annotations in this validation if the annotations are mostly defining things on data model objects?